Improve view in Solution PM UI for Solution Explorer gold bar entry point

jebriede opened the pull request:
<!-- DO NOT MODIFY OR DELETE THIS TEMPLATE. IT IS USED IN AUTOMATION. -->

# Bug

<!-- If this is an engineering change or test change only, you do not need an issue. -->
<!-- Find or create an issue in NuGet/Home and paste the full url. -->
<!-- At the maintainers discretion, multiple changes may apply to a single issue, but only when the PRs are all created within a short period of time. -->
Fixes: https://github.com/NuGet/Home/issues/13050

## Description
Open the Solution PM UI to the Installed tab with the Show Only Vulnerable packages filter checked when entering the PM UI from the gold bar in the Solution Explorer that warns of vulnerable packages in the solution.

To accomplish this, added launch options to the existing Solution PM UI command handler that supports opening the PM UI to a specific tab and with specific filters on, off, or with default values.

## PR Checklist

- [x] Meaningful title, helpful description and a linked NuGet/Home issue
- [ ] ~~Added tests~~
- [ ] ~~Link to an issue or pull request to update docs if this PR changes settings, environment variables, new feature, etc.~~

Manual validation of scenario:
Clicking the Manage NuGet Package link in the Solution Explorer will take you to this view:
<img width="1919" height="767" alt="image" src="https://github.com/user-attachments/assets/b7e35c57-1b3f-48df-bade-8ba3887b2481" />

I verified that this view opens even when the Solution PM UI had not previously been opened and thus the extension hadn't been preloaded.

If the Solution PM UI is already open, clicking this link will switch to the Installed tab with the Show only vulnerable packages option checked.

Also, regression tested the menu item by right clicking the Solution -> Manage NuGet Packages for Solution... as well as Tools -> NuGet Package Manager -> Manage NuGet Packages for Solution...

jebriede committed:
Add new command to launch solution PM UI to the Installed Tab with the Vulnerabilities filter on

nkolev92 commented on src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs:
```diff
@@ -378,6 +378,12 @@ private async Task AddMenuCommandHandlersAsync()
                 _mcs.AddCommand(updatePackagesDialogCommand);
+
+                // menu command for launching the PM UI for the solution to show vulnerable packages
+                var launchPackageManagerUICommandID = new CommandID(GuidList.guidNuGetDialogCmdSet, PkgCmdIDList.cmdidLaunchPackageManagerUI);
```
nit: seeing lots of this repeated code makes me think that we can extra helper method for this olemenu command creation that'd prevent accidental mistakes.

Nothing to fix now, but I know you look at the code more, so maybe that gives you an idea for the future.,

jebriede replied:
Refactored quite a lot and removed this new command. Your comment is still relevant and noted for future refactoring opportunities, but this PR no longer introduces a new command.

Nigusu-Allehu commented on src/NuGet.Clients/NuGet.Tools/PackageManagerLaunchService.cs:
```diff
@@ -28,10 +28,12 @@ public void LaunchSolutionPackageManager()
         }
     }
+
+    // PkgCmdIDList.cmdidAddPackageDialogForSolution,
```
do we wanna remove this

donnie-msft replied:
👍

jebriede replied:
Removed, thanks!

donnie-msft reviewed (dismissed):
Code looks good. Would be nice to have a screenshot and to describe the validation a bit as well.

Frulfump commented:
This was closed but I was really looking forward to some better UX/DevEx in this area, there was talk about things improving for VS 2026 is that still the case without this PR?

jebriede committed:
Added launch options to existing Solution PM UI command handler

jebriede commented:
> This was closed but I was really looking forward to some better UX/DevEx in this area, there was talk about things improving for VS 2026 is that still the case without this PR?

@Frulfump thanks for the feedback! This PR was closed automatically by a bot. I've reopened it now with improved changes. :-)

jebriede commented:
> Code looks good. Would be nice to have a screenshot and to describe the validation a bit as well.

@donnie-msft thanks! I've refactored the PR quite a bit and, in this iteration, I've added more details about the validation and a screenshot thanks to your suggestion.
